### PR TITLE
Suspend all BMC/Compuware plugins

### PR DESCRIPTION
**Pull request by daniel-beck**
We got a complaint via GitHub Trust & Safety about code copyrighted by BMC Software Inc in one of our plugin repos, specifically https://github.com/jenkinsci/compuware-common-configuration-plugin/blob/85e15ffc9a2987453f61a1f41f91d7fe109d5636/src/test/java/com/compuware/jenkins/common/configuration/HostConnectionTest.java#L56 and https://github.com/jenkinsci/compuware-common-configuration-plugin/blob/85e15ffc9a2987453f61a1f41f91d7fe109d5636/src/test/java/com/compuware/jenkins/common/configuration/HostConnectionTest.java#L256 (yes, really).

Out of an abundance of caution I recommend we suspend all of their plugins' distribution.

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
cc @thorveshubham and @pramodc0777

**Comment by pramodc0777** [comment]
@daniel-beck  apologies  for those issues,I would  request you not to suspend  those plugins,we will  fix and push this common plugin  by tomorrow.Please  consider  this request Thanks.

**Comment by timja** [comment]
Thanks, the fully copy of the email we received is:

> Hello,
> 
> I'm reaching out on behalf of the GitHub Trust & Safety Team to let you know we received a report that one of your repositories contains private information that was posted without consent. Specifically, the following content was reported:
>
> https://github.com/jenkinsci/compuware-common-configuration-plugin/blob/85e15ffc9a2987453f61a1f41f91d7fe109d5636/src/test/java/com/compuware/jenkins/common/configuration/HostConnectionTest.java
> As it have copyright source code/password of BMC Software Inc
> lines 56 and 256
>
> In order to remove the content in question, we ask that you refer to the following article for help:
>
> [Removing sensitive data from a repository](https://docs.github.com/articles/remove-sensitive-data)
>
> Please make sure to follow those instructions carefully, as simply deleting the content will not remove it completely from the repository commit history.
>
> Alternately, you may simply want to switch the repository to private by following the instructions found here:
>
> [Making a repository private](https://docs.github.com/en/github/administering-a-repository/setting-repository-visibility#making-a-repository-private)
>
> If these changes are not made within 3 business days, we will continue our review of the complaint. We may need to disable your repository at that time in order to protect the owner of private information that has been posted in violation of our [Acceptable Use Policies](https://docs.github.com/en/github/site-policy/github-acceptable-use-policies#3-conduct-restrictions).
>
> If you have any questions, concerns, or feedback regarding this notice, please let us know as soon as possible.
>
> Regards, 
> GitHub Trust & Safety

**Comment by daniel-beck** [comment]
@pramodc0777 I recommend you check with the folks at BMC who submitted this false report to GitHub so this doesn't happen again. Or is the source code not, in fact, published under an MIT license by you?

**Comment by pramodc0777** [comment]
@daniel-beck & @timja we have added the resolution for this issue and removed commit history for this file too from GitHub, please consider this fix to avoid suspension of all plugins.

Thanks.

**Comment by daniel-beck** [comment]
This does not actually resolve the problem.

We need to know whether it is legal for us to host and distribute these plugins. The copyright claim indicates it is not.

**Comment by MarkEWaite** [comment]
The source code of `HostConnectionTest.java` was added to the repository 7 Jul 2017 by author `PFHJYG0` with a compuware.com email address using commit 7bcc4256f475b0b4acc4928c0760dbd1ad7c90e7 .  It included the MIT license and a 2017 copyright date for Compuware.  

@daniel-beck is your concern that the originally claimed MIT license on the file is not valid?  

The GitHub Trust & Safety Team message says

> contains private information that was posted without consent

I thought that meant the private information inside the file was their concern, not the copyright on the file.  I thought that the later phrasing that says:

> As it have copyright source code/password of BMC Software Inc
> lines 56 and 256

was offering the specific details of the private information that was posted without consent.  I didn't interpret that phrasing as a claim that the license in the source file (MIT) or the copyright owner (Compuware) were invalid or prevented redistribution.  If the MIT license on the source file is valid, then I believe that allows us to distribute the plugin and its source code.  If the MIT license on the source file is not valid, then that invalid license was already placed there in 2017 with the original creation of the file.

Can you help me understand the concern about copyright and the right to distribute the plugin?

I've placed this on the governance board agenda for today so that board members are aware of the concern.

**Comment by MarkEWaite** [comment]
I sent an email message to the committers on `HostConnectionTest.java`  asking for their help to identify the bot or person that submitted the report to the GitHub Trust & Safety Team.  The message bounced for 3 of the 5 recipients with "recipient address rejected".  I hope the other two people will respond here.

**Comment by daniel-beck** [comment]
A lot of Compuware stuff was recently re-licensed under a non-free license:

https://github.com/jenkinsci/compuware-common-configuration-plugin/blob/master/LICENSE
https://github.com/jenkinsci/compuware-ispw-operations-plugin/blob/master/LICENSE.txt
https://github.com/jenkinsci/compuware-strobe-measurement-plugin/blob/master/LICENSE.txt
https://github.com/jenkinsci/compuware-scm-downloader-plugin/blob/master/LICENSE.txt

@pramodc0777 @vbugale @vikpatil270 Please note that a requirement for plugin hosting is that plugins must be published under an OSI-approved open source license.

**Comment by oleg-nenashev** [comment]
I am +1 for depublishing is if an OSI-compliant license is not restored for
the plugins, and for all upstream dependencies if any. The hosting
requirements are clear in this regards, and probably we need some GitHub
org automation to raise the flag for maintainers right away.

**Comment by daniel-beck** [comment]
As discussed in [today's project meeting](https://docs.google.com/document/d/11Nr8QpqYgBiZjORplL_3Zkwys2qK1vEvK-NYyYa4rzg/edit#heading=h.sn91mb5c7g3x), we're suspending all BMC / Compuware plugins, pending confirmation that GitHub Trust & Safety claims have been rescinded and no future claims will be made, and restoration of OSI compatible license (where applicable, see https://github.com/jenkins-infra/repository-permissions-updater/pull/3201)

**Comment by daniel-beck** [comment]
I've made https://github.com/jenkinsci/compuware-common-configuration-plugin private and downgraded Admin permissions to Write for all current collaborators to prevent them from changing repository status. This should take care of the specific GitHub complaint while the above is being implemented.

**Comment by pramodc0777** [comment]
@daniel-beck what is the process  if we restored  or changed  to oci compliant license to get plugins  published  again?

**Comment by MarkEWaite** [comment]
https://groups.google.com/g/jenkinsci-board/c/vHbqCK7ywuw was sent to the maintainers of the affected plugins as noted in the repository permissions updater.  It includes instructions for what needs to be changed in order for the plugins to be published again.

That message says:

Late last week the Jenkins project received a complaint filed by BMC Software Inc. with GitHub Trust & Safety that the Compuware Common Configuration plugin for Jenkins contained private information that was posted without consent.


The content that was reported was lines 56 and 256 of the HostConnectionTest.java source file at SHA1 85e15ffc9a2987453f61a1f41f91d7fe109d5636 .  


Line 56 has the text:

```

private static final String CERT_PASSWORD = "compuware";
```


Line 256 has the text:


```
/**

```

Yes, one of the offending lines reported is literally the start of a Javadoc comment and the other is the word “compuware” as a variable named “CERT_PASSWORD”.


The GitHub Trust & Safety team allows 3 business days to resolve the complaint and states that they “may need to disable your repository [..] in order to protect the owner of private information”.  The Jenkins project depends on GitHub services for source control.  We cannot risk that they will disable or reduce our access.


The repository of the Compuware Common Configuration plugin for Jenkins has been made private based on the recommendation from GitHub Trust & Safety.  Distribution of the Compuware Common Configuration plugin has been suspended by the Jenkins project until it has been confirmed that the GitHub Trust & Safety claims have been rescinded and no future claims will be made.


While researching the claim from GitHub Trust & Safety, we found that four Jenkins plugins from Compuware have been modified to use a software license that is not an OSI approved open source license.  The Jenkins project only publishes free and open source plugins distributed under an [OSI-approved](https://opensource.org/licenses/) license.


We’ve blocked futured releases of Compuware plugins that now have licenses that are not OSI approved open source licenses.  Those plugins are:


* compuware-scm-downloader
* compuware-ispw-operations
* compuware-common-configuration
* compuware-strobe-measurement


New releases of those plugins will remain blocked until they are licensed under an OSI approved open source license.


We found that two of the four Jenkins plugins that use a software license that is not an OSI approved open source license have been released and those releases have been stored in the Jenkins artifact repository.  The Jenkins project only publishes free and open source plugins distributed under an [OSI-approved](https://opensource.org/licenses/) license.


We’ve removed the following plugin releases from the Jenkins artifact repository due to their use of a license that is not an OSI approved open source license:


* compuware-ispw-operations 1.0.12, 2.0.0, 2.1.0
* compuware-scm-downloader 2.0.15 and 2.0.16

Based on the claim to GitHub Trust & Safety and the change of license on the four Compuware plugins, the Jenkins project governing board approved on 20 Mar 2023 that all Jenkins plugins from BMC Software Inc. should be suspended from distribution until it has been confirmed that the claims to GitHub Trust & Safety have been rescinded, that no future claims will be made, and that the OSI approved licenses will be restored on the plugins that have had their licenses changed.  The plugins that have been suspended are:


* bmc-cfa
* bmc-change-manager-imstm
* bmc-rpd
* compuware-common-configuration
* compuware-ispw-operations
* compuware-scm-downloader
* compuware-strobe-measurement
* compuware-topaz-for-enterprise-data
* compuware-topaz-for-total-test
* compuware-topaz-utilities
* compuware-xpediter-code-coverage
* compuware-zadviser-api

We hope that you will take the necessary actions to allow the Jenkins plugins from BMC Software Inc. to again be distributed by the Jenkins project.


Mark Waite

Member of the Jenkins Board

**Comment by MarkEWaite** [comment]
@pramodc0777 

> @daniel-beck what is the process if we restored or changed to oci compliant license to get plugins published again?

I hope that the [message above](https://github.com/jenkins-infra/update-center2/pull/692#issuecomment-1477267700) answers your question on the process to have the plugins published again.  If not, please reply in the comments here and we can work through the detailed steps that are needed.

**Comment by pramodc0777** [comment]
Yes @MarkEWaite we are working  on it.

**Comment by pramodc0777** [comment]
Hi @MarkEWaite  @daniel-beck  @timja  We have restored licenses to MIT for all 4 plugins.
- compuware-scm-downloader
- compuware-ispw-operations
- compuware-common-configuration
- compuware-strobe-measurement

Also we have fixed Issues with  HostConnectionTest.java which was mentioned previously, would you please review it and let us know if anything needs to be done for republishing plugins.

**Comment by timja** [comment]
This step here:
> until it has been confirmed that the claims to GitHub Trust & Safety have been rescinded, that no future claims will be made

**Comment by pramodc0777** [comment]
@timja would you help on how can we do that, as we have already fixed whatever is asked in  GitHub Trust & Safety Team email you shared to us, also can we get information about report like who reported this that will be helpful.

**Comment by timja** [comment]
I've asked

**Comment by timja** [comment]
They replied with:

 Hello,

For privacy and security reasons, we’re unable to provide information about other users’ accounts unless presented with appropriate legal process.

Please see the article below for more details: 
https://docs.github.com/en/github/site-policy/guidelines-for-legal-requests-of-user-data

**Comment by MarkEWaite** [comment]
@pramodc0777 I think that we need a statement from whatever organization at BMC raised that issue with GitHub Trust & Safety.  The statement should tell us that the issue they raised is resolved and that they do not intend to raise any further issues with other plugins maintained by employees of BMC Software Inc.

Based on the quote from GitHub Trust & Safety, I don't think that GitHub will provide the information about whoever raised that issue.  I think that you'll need to find the reporting person or entity by a search inside BMC.  I don't envy you in doing that search, since BMC is a diverse and distributed company.

If you can't find that person or system inside BMC that initiated the original report, I think that it will need to be brought to a Jenkins Board meeting to see if the Board is willing to relax that requirement.

**Comment by daniel-beck** [comment]
I asked GitHub support for confirmation.

**Comment by daniel-beck** [comment]
> Thanks for reaching out to us.
> 
> For privacy and security reasons, we do not discuss our review process or the results of a specific investigation. It is up to your discretion as to the visibility of your repository.

So GitHub is not helpful. As far as I'm concerned (others may see this differently), the complaint is still active unless and until GitHub responds to us confirming that your claim is rescinded. I suggest you figure out how to get GitHub to confirm this.

**Comment by daniel-beck** [comment]
It's unclear what you're asking. We'll take care of restoring artifact, publication of plugins, and upload permissions once GitHub confirms this is resolved (or someone else at the project who has more relaxed requirements than me steps up). How would a discussion forum help?

**Comment by daniel-beck** [comment]
@cpwr-devops We don't have any access to GitHub besides their support channel. They respond to our requests with boilerplate answers. I suggest you contact them through the same method you used to send your complaint in the first place.

**Comment by daniel-beck** [comment]
We got a response from GitHub:

> I’m writing to let you know that the requestor of the following content has retracted their private information removal request:
> 
> https://github.com/jenkinsci/compuware-common-configuration-plugin/blob/85e15ffc9a2987453f61a1f41f91d7fe109d5636/src/test/java/com/compuware/jenkins/common/configuration/HostConnectionTest.java
> 
> As such, we will not be taking any further action on your content at this time.

I plan to act on this tomorrow, restoring repos, artifacts, plugin publishing to their previous states.

**Comment by daniel-beck** [comment]
I made the GH repo public and restored previous repo admin permissions. Note that our taking the repository private detached your forks from it (e.g., https://github.com/vaibhavjain2/compuware-common-configuration-plugin is now a fork of https://github.com/tkosacek/CPWR-Common and not https://github.com/jenkinsci/compuware-common-configuration-plugin, which no longer has any forks). This is not something we can fix.

https://github.com/jenkins-infra/repository-permissions-updater/pull/3218 restores the ability to upload new releases once merged.
https://github.com/jenkins-infra/update-center2/pull/693 restores plugin distribution (plugins.jenkins.io and plugin manager) once merged.

**Comment by daniel-beck** [comment]
Yes, which is expected. Read my previous comment again, carefully.

**Comment by pramodc0777** [comment]
@daniel-beck  it should get republished automatically as you already merged pr for both #693 ,#3218 , do we need to make new release to get plugin on (plugins.jenkins.io and plugin manager).

**Comment by daniel-beck** [comment]
> as you already merged pr for both https://github.com/jenkins-infra/update-center2/pull/693

The PR is still open.

**Comment by daniel-beck** [comment]
Merged #693. Note that as a result of the exclusion of the improperly licensed releases, the currently newest releases of two of your plugins will remain unavailable. I recommend you look into publishing new releases with a fixed license.

**Comment by daniel-beck** [comment]
Yes, see Mark's email.

**Pull request merged**
